pkg/proxy/util/iptables/traffic.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iptables
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/klog"
+	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
+	utilnet "k8s.io/utils/net"
+)
+
+// LocalTrafficDetector in a interface to take action (jump) based on whether traffic originated locally
+// at the node or not
+type LocalTrafficDetector interface {
+	// IsImplemented returns true if the implementation does something, false otherwise
+	IsImplemented() bool
+
+	// JumpIfLocal appends conditions to jump to a target chain if traffic detected to be
+	// of local origin
+	JumpIfLocal(args []string, toChain string) []string
+
+	// JumpINotfLocal appends conditions to jump to a target chain if traffic detected not to be
+	// of local origin
+	JumpIfNotLocal(args []string, toChain string) []string
+}
+
+type noOpLocalDetector struct{}
+
+// NewNoOpLocalDetector is a no-op implementation of LocalTrafficDetector
+func NewNoOpLocalDetector() LocalTrafficDetector {
+	return &noOpLocalDetector{}
+}
+
+func (n *noOpLocalDetector) IsImplemented() bool {
+	return false
+}
+
+func (n *noOpLocalDetector) JumpIfLocal(args []string, toChain string) []string {
+	return args // no-op
+}
+
+func (n *noOpLocalDetector) JumpIfNotLocal(args []string, toChain string) []string {
+	return args // no-op
+}
+
+type detectLocalByCIDR struct {
+	cidr string
+}
+
+// NewDetectLocalByCIDR implements the LocalTrafficDetector interface using cluster CIDR.
```

Comment:
This should be more complete.  This uses a cluster CIDR to approximate "this node" but that assumes that we will never have to consider traffic from other nodes in teh cluster in the same circumstance

Reply:
You mean the comment? Good catch. Yes, the CIDR here can either be cluster CIDR or the node CIDR. So after this refactor, it's mostly plumbing of the node CIDR using the same implementation and adding tests. Will update the comment. Anything more you specifically think is worth capturing in addition to the comment? Should we add a link to the KEP for example for readers to understand the reason why?